src/main/java/frc/robot/subsystems/drive/SparkMaxOdometryThread.java

```diff
@@ -83,11 +83,22 @@ private void periodic() {
     Drive.odometryLock.lock();
     double timestamp = Logger.getRealTimestamp() / 1e6;
     try {
+      double[] values = new double[signals.size()];
+      boolean isValid = true;
       for (int i = 0; i < signals.size(); i++) {
-        queues.get(i).offer(signals.get(i).getAsDouble());
+        OptionalDouble value = signals.get(i).get();
```

Comment:
same as above..?

Reply:
No it's because you need to check if it is valid or not